.github/workflows/on_main_push.yml

```diff
@@ -1,137 +0,0 @@
-name: Build & Deploy
-
-on:
-  workflow_dispatch:
-  push:
-    branches:
-      - main
-      - trailofbits
-
-jobs:
-  run-check:
-    uses: ./.github/workflows/run-checks.yml
-  run-benchmarks:
-    uses: ./.github/workflows/run-benchmarks.yml
-  build-deploy-docs:
-    uses: ./.github/workflows/build-deploy-docs.yml
-    with:
-      deploy: ${{ github.ref == 'refs/heads/main' }}
-  build-linux:
-    strategy:
-      fail-fast: false
-      matrix:
-        name:
-          - openssl111
-          - openssl111j
-          - openssl101f_asan
-          - openssl102u
-          - libressl
-          - wolfssl430
-          - wolfssl510
-          - wolfssl510-sig
-          - wolfssl510-skip
-          - wolfssl520
-          - wolfssl520_asan
-          - wolfssl530
-          - wolfssl530_asan
-          - wolfssl540
-          - wolfssl540-sdos2
-          - wolfssl530-cdos
-          - wolfssl540_asan
-          - wolfssl540-buf
-          - wolfssl540-heap
-          - wolfssl540_asan-perf
-          - wolfssl540-perf
-          - libssh
-          - tlspuffin_no_default_features
```

Comment:
It seems like we are no longer building the tlspuffin binaries in CI or did I miss a file?

Building binaries is important additionally to tests to catch issues with that.

Reply:
No, we still build them, but the entry point is now in the `ci.yml` workflow. The test matrix has not changed in this PR.